pandas/core/dtypes/common.py

```diff
@@ -319,6 +319,13 @@ def is_datetime64tz_dtype(arr_or_dtype) -> bool:
     >>> is_datetime64tz_dtype(s)
     True
     """
+    # GH#52607
+    warnings.warn(
+        "is_datetime64tz_dtype is deprecated and will be removed in a future "
```

Comment:
This could be useful if we want to check whether it's a numpy or arrow tz type, but idk if we have a use case for that yet.

Reply:
It could. I think the fact that it is ambiguous (based on the name) is all the more reason to deprecate it